code/modules/mob/living/simple_animal/abnormality/he/silent_girl.dm

```diff
@@ -32,98 +33,77 @@
 	gift_type = /datum/ego_gifts/remorse
 	abnormality_origin = ABNORMALITY_ORIGIN_ARTBOOK
 
-
-/mob/living/simple_animal/hostile/abnormality/silent_girl/proc/OnWorkStart(datum/source, datum/abnormality/datum_reference, mob/living/carbon/human/user, work_type)
-	SIGNAL_HANDLER
-	if(user.stat == DEAD)
-		return FALSE
-	if(work_type != ABNORMALITY_WORK_ATTACHMENT)
-		return FALSE
-	if(datum_reference == src.datum_reference)
-		return FALSE
-	if(user.has_status_effect(STATUS_EFFECT_SG_GUILTY))
-		user.remove_status_effect(STATUS_EFFECT_SG_GUILTY)
-		datum_reference.qliphoth_change(1)
-	else if(user.has_status_effect(STATUS_EFFECT_SG_ATTONEMENT))
-		user.remove_status_effect(STATUS_EFFECT_SG_ATTONEMENT)
-
-/mob/living/simple_animal/hostile/abnormality/silent_girl/proc/GuiltEffect(mob/living/carbon/human/user)
-	if ((user.stat == DEAD) || (user.has_status_effect(STATUS_EFFECT_SG_GUILTY)))
+/mob/living/simple_animal/hostile/abnormality/silent_girl/proc/GuiltEffect(mob/living/carbon/human/user, enable_qliphoth = TRUE, stack_count = 1)
+	if (user.stat == DEAD)
 		return
-	datum_reference.qliphoth_change(-1)
-	user.apply_status_effect(STATUS_EFFECT_SG_GUILTY, datum_reference)
+	if(enable_qliphoth)
+		datum_reference.qliphoth_change(-1, user)
+		user.apply_status_effect(STATUS_EFFECT_SG_GUILTY, datum_reference, stack_count)
+	else
+		user.apply_status_effect(STATUS_EFFECT_SG_GUILTY, null, stack_count)
 	return
 
 /mob/living/simple_animal/hostile/abnormality/silent_girl/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
-	if(get_attribute_level(user, PRUDENCE_ATTRIBUTE) < 60 && !user.has_status_effect(STATUS_EFFECT_SG_GUILTY))
+	if(get_attribute_level(user, PRUDENCE_ATTRIBUTE) < 60)
 		GuiltEffect(user)
 	return
 
 /mob/living/simple_animal/hostile/abnormality/silent_girl/AttemptWork(mob/living/carbon/human/user, work_type)
-	if (user.has_status_effect(STATUS_EFFECT_SG_GUILTY) || user.has_status_effect(STATUS_EFFECT_SG_ATTONEMENT))
-		user.apply_status_effect(STATUS_EFFECT_SG_ATTONEMENT, datum_reference) //If a guilty person works on her, they panic.
+	if (user.has_status_effect(STATUS_EFFECT_SG_GUILTY))
```

Comment:
If you want my opinion silent girl should just kill you if you do this. That's a bit closer to the lore at least.